src/pynxtools/dataconverter/readers/multi/reader.py

```diff
@@ -0,0 +1,446 @@
+    """
+
+    # Whitelist for the NXDLs that the reader supports and can process
+    supported_nxdls: List[str] = []
+    extensions: Dict[str, Callable[[Any], dict]] = {}
+    kwargs: Optional[Dict[str, Any]] = None
+    overwrite_keys: bool = True
+    processing_order: Optional[List[str]] = None
+    config_file: Optional[str] = None
+    config_dict: Dict[str, Any]
+
+    def __init__(self, config_file: Optional[str] = None):
+        self.callbacks = ParseJsonCallbacks(
+            attrs_callback=self.get_attr,
+            data_callback=self.get_data,
+            eln_callback=self.get_eln_data,
+            dims=self.get_data_dims,
+        )
+        self.config_file = config_file
+        self.config_dict = {}
+
+    def setup_template(self) -> Dict[str, Any]:
+        """
+        Setups the initial data in the template.
+        This may be used to set fixed information, e.g., about the reader.
+        """
+        return {}
+
+    # pylint: disable=unused-argument
+    def handle_objects(self, objects: Tuple[Any]) -> Dict[str, Any]:
+        """
+        Handles the objects passed into the reader.
+        """
+        return {}
+
+    def get_attr(self, key: str, path: str) -> Any:
+        """
+        Returns an attributes from the given path.
+        Should return None if the path does not exist.
+        """
+        return None
+
+    def get_data(self, key: str, path: str) -> Any:
+        """
+        Returns data from the given path.
+        Should return None if the path does not exist.
+        """
+        return None
+
+    def get_eln_data(self, key: str, path: str) -> Any:
+        """
+        Returns data from the given eln path.
+        Should return None if the path does not exist.
+        """
+        return {}
+
+    def get_data_dims(self, key: str, path: str) -> List[str]:
+        """
+        Returns the dimensions of the data from the given path.
+        """
+        return []
+
+    def get_entry_names(self) -> List[str]:
+        """
+        Returns a list of entry names which should be constructed from the data.
+        Defaults to creating a single entry named "entry".
+        """
+        return ["entry"]
+
+    def post_process(self) -> None:
+        """
+        Do postprocessing after all files and config file are read.
+        """
+
+    def read(
+        self,
+        template: dict = None,
+        file_paths: Tuple[str] = None,
+        objects: Tuple[Any] = None,
+        **kwargs,
+    ) -> dict:
+        """
+        Reads data from multiple files and passes them to the appropriate functions
+        in the extensions dict.
+        """
+        self.kwargs = kwargs
+        self.config_file = self.kwargs.get("config_file", self.config_file)
+        self.overwrite_keys = self.kwargs.get("overwrite_keys", self.overwrite_keys)
+
+        template = Template(overwrite_keys=self.overwrite_keys)
```

Comment:
For my tests, this created an empty template. Not sure if this is what it's supposed to do.

Reply:
That's correct. This is the initialisation of the template and should just create an empty template